src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -668,8 +675,13 @@ static bool pick_from_internal_rr_locked(
 
 static grpc_lb_policy_args *lb_policy_args_create(grpc_exec_ctx *exec_ctx,
                                                   glb_lb_policy *glb_policy) {
-  grpc_lb_addresses *addresses =
-      process_serverlist_locked(exec_ctx, glb_policy->serverlist);
+  grpc_lb_addresses *addresses;
+  if (glb_policy->fallback) {
+    addresses = glb_policy->fallback_addresses;
```

Comment:
It's not clear from the code what the ownership semantics are for `glb_policy->fallback_addresses`.  Here we are giving the value to the local variable `addresses`, whose value will be freed before this function returns.  That will leave us in a state where `glb_policy->fallback_addresses` points to a value that has already been freed.

If we intend it to be freed at this point, then let's reset `glb_policy->fallback_addresses` to NULL right after this line, so that it's clear that nothing should try accessing it later.

If we do *not* intend it to be freed at this point, then we should change line 696 to only free the value if we are not using the fallback.  In that case, we should make sure to free the value in `glb_destroy()`.